pkg/objectstore/bucket.go

```diff
@@ -191,21 +190,7 @@ func s3BucketRegion(ctx context.Context, cfg ProviderConfig, sec Secret, bucketN
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to create session, region = %s", r)
 	}
-	cli := s3.New(s)
-	if cli == nil {
-		return "", errors.New("failed to create s3 client")
-	}
-	gbli := &s3.GetBucketLocationInput{
-		Bucket: aws.String(bucketName),
-	}
-	gblo, err := cli.GetBucketLocation(gbli)
-	if err != nil {
-		return "", errors.Wrap(err, "failed to get bucket location")
-	}
-	if gblo.LocationConstraint != nil {
-		return *gblo.LocationConstraint, nil
-	}
-	return "", nil
+	return s3manager.GetBucketRegion(ctx, s, bucketName, cfg.Region)
```

Comment:
What's the reasoning behind switching to `GetBucketRegion`?

Reply:
They use slightly different auth schemes. I think I'd prefer to move to GetBucketLocation, but there are several tests that assume our behavior is based on GetBucketRegion. 

@miquella was also concerned about cross region support using GetBucketLocation. We should probably try the instructions in this doc to verify: https://docs.aws.amazon.com/AmazonS3/latest/dev/example-walkthroughs-managing-access-example2.html